models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb

```diff
@@ -0,0 +1,321 @@
+   },
+   "outputs": [],
+   "source": [
+    "model.fit(X_train, y_train, epochs=epochs, verbose=0.5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Evaluate the model metrics.\n",
+    "\n",
+    "After the model is trained, evaluate the model against the test set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Compute the predictions (y_predictions) given the test data (y_predicted)\n",
+    "y_predicted_probabilities = model.predict(X_test)\n",
+    "y_predicted = tf.argmax(y_predicted_probabilities, axis=1)\n",
+    "\n",
+    "# Compare the predicted values for the test set (y_predicted)\n",
+    "# against the expected values (y_test)\n",
+    "print(\"Classification Report:\")\n",
+    "print(classification_report(y_test, y_predicted))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The trained model has an accuracy value of aproximately 79%.\n",
+    "\n",
+    "You can improve the score by retraining the model after more sophisticated data engineering or by tweaking the model hyper parameters."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 6. Test the model with sample cases.\n",
+    "Test the model with data from two patients: one patient with diabetes and one patient without diabetes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Tuple for textual display of prediction\n",
+    "classes = ('No diabetes', 'Diabetes')\n",
+    "\n",
+    "\n",
+    "def predict(patients: List[Dict]):\n",
+    "    features_as_lists = [list(patient.values()) for patient in patients]\n",
+    "    inputs_array = np.array(features_as_lists)\n",
+    "    prediction_probabilities = model.predict(inputs_array, verbose=0)\n",
+    "    # argmax gets the index of the maximum value in an array\n",
+    "    predictions = [classes[np.argmax(p)] for p in prediction_probabilities]\n",
+    "    return predictions\n",
+    "\n",
+    "\n",
+    "diabetes_patient = {\n",
+    "    \"Pregnancies\": 6.0,\n",
+    "    \"Glucose\": 110.0,\n",
+    "    \"BloodPressure\": 65.0,\n",
+    "    \"SkinThickness\": 15.0,\n",
+    "    \"Insulin\": 1.0,\n",
+    "    \"BMI\": 45.7,\n",
+    "    \"DiabetesPedigreeFunction\": 0.627,\n",
+    "    \"Age\": 50\n",
+    "}\n",
+    "\n",
+    "no_diabetes_patient = {\n",
+    "    \"Pregnancies\": 0,\n",
+    "    \"Glucose\": 88.0,\n",
+    "    \"BloodPressure\": 60.0,\n",
+    "    \"SkinThickness\": 35.0,\n",
+    "    \"Insulin\": 1.0,\n",
+    "    \"BMI\": 45.7,\n",
+    "    \"DiabetesPedigreeFunction\": 0.27,\n",
+    "    \"Age\": 20\n",
+    "}\n",
+    "\n",
+    "predictions = predict([diabetes_patient, no_diabetes_patient])\n",
+    "print(predictions)"
```

Comment:
With a 79% model accuracy I always get `Diabetes, Diabetes` as the prediction result.

Reply:
Fixed, now the prediction should be correct and reproducible.